starknet_py/hash/casm_class_hash.py

```diff
@@ -10,40 +8,62 @@
     BytecodeSegmentStructure,
     NestedIntList,
 )
+from starknet_py.hash.hash_method import HashMethod
 from starknet_py.net.client_models import CasmClassEntryPoint
 from starknet_py.net.executable_models import CasmClass
 
 CASM_CLASS_VERSION = "COMPILED_CLASS_V1"
 
 
-def compute_casm_class_hash(casm_contract_class: CasmClass) -> int:
+def get_casm_hash_method_for_rpc_version(rpc_version: str) -> HashMethod:
+    major, minor, patch = _parse_version(rpc_version)
+
+    # RPC 0.10.0 and later use Blake2s
+    if (major, minor, patch) >= (0, 10, 0):
+        return HashMethod.BLAKE2S
+
+    return HashMethod.POSEIDON
+
+
+def _parse_version(version: str) -> Tuple[int, int, int]:
+    parts = version.split(".")
+
+    if len(parts) != 3:
+        raise ValueError("Version must have three parts separated by dots.")
+
+    return int(parts[0]), int(parts[1]), int(parts[2])
```

Comment:
Actually, let's get rid of this logic from our codebase and use https://pypi.org/project/semver/